### ISSUE TITLE
Unable to run tests when they use different cookies

### ISSUE THREAD

**Issue by MartijnVanRiele** [cypress-io/cypress#484]
### Current behavior:

I have like 5 different spec-files. With the spec i test different behavior of our site. To test this behavior we have to use different users that are logged in to the site. To do this, i set cookies of those user at the begin of the spec, so i can avoid to go through the login-page.

For instance:

```js
describe('Payment Page (Betalen)', () => {
  beforeEach(() => {
    cy.clearCookies();
    //klantnummer = 24562362
    cy.setCookie(
      'token', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJqdGkiOiIwNWE2YmY4OS0yZjE3LTRmZWYtYjI1Mi05NDAxY2ZjZWFlZmMiLCJhdXQiOiJmdWxsIiwicHJlX2F1dGhfc2hvcHBlciI6ImVlYmI0M2VlLTgzOWEtNDViNS05Njg1LTBmNDAyYzBjNDUxMCIsImlhdCI6MTQ5MDYwNTQwMiwic3ViIjoiMjQ1NjIzNjIiLCJpc3MiOiJibGF6ZS1nYXRld2F5IiwiaXNzX3JlYXNvbiI6ImxvZ2luIn0.acudOmfCFP5CzKAHD5KQYFeLFuBSoXDtstJgK11arqw',
      {
        path: '/',
        domain: '.nl.wehkamp.dev.blaze-public.ps',
      }
    );

    cy.setCookie(
      'user-data',
      'eyJhdXRoZW50aWNhdGlvbl90eXBlIjoiZnVsbCIsInRva2VuX2lkIjoiMDVhNmJmODktMmYxNy00ZmVmLWIyNTItOTQwMWNmY2VhZWZjIn0=',
      {
        path: '/',
        domain: '.nl.wehkamp.dev.blaze-public.ps',
      }
    );

    cy.visit('/gespreid-betalen/betalen');
  });
```

So these 5 specfiles each have a different cookie. 

When i run these specs individually, it all runs ok. When i us the 'Run All Tests'-function, the first test runs ok but the second (which uses another cookie) doesn't. I get redirected to the loginpage.

### Expected behavior:

That i can run all tests in 1 run, each test using different `setCookie`

### Additional Info (images, notes, stack traces, etc)

* Operating System: OS
* Cypress Version: 0.19.1

**Comment by jennifer-mann** [comment]
I was unable to recreate the issue above. I created a repo below that closely resembles the test code you described above. So, I have a few follow-up suggestions:

1. What happens if you write the assertion below in the first test of each file (substituting the values for the appropriate token & user-data), then click "Run All". Do these assertions pass? fail?

```js
it('gets the browser cookie', function(){
    cy.getCookie('token').should('have.property', 'name', 'token')
    cy.getCookie('token').should('have.property', 'value', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJqdGkiOiIwNWE2YmY4OS0yZjE3LTRmZWYtYjI1Mi05NDAxY2ZjZWFlZmMiLCJhdXQiOiJmdWxsIiwicHJlX2F1dGhfc2hvcHBlciI6ImVlYmI0M2VlLTgzOWEtNDViNS05Njg1LTBmNDAyYzBjNDUxMCIsImlhdCI6MTQ5MDYwNTQwMiwic3ViIjoiMjQ1NjIzNjIiLCJpc3MiOiJibGF6ZS1nYXRld2F5IiwiaXNzX3JlYXNvbiI6ImxvZ2luIn0.acudOmfCFP5CzKAHD5KQYFeLFuBSoXDtstJgK11arqw')
    cy.getCookie('user-data').should('have.property', 'name', 'user-data')
    cy.getCookie('user-data').should('have.property', 'value', 'eyJhdXRoZW50aWNhdGlvbl90eXBlIjoiZnVsbCIsInRva2VuX2lkIjoiMDVhNmJmODktMmYxNy00ZmVmLWIyNTItOTQwMWNmY2VhZWZjIn0=')
})
```

2. Fork the repo below and get their tests to fail. There is something different that I haven't determined. This would be *very* helpful to us solving the issue.

**Test Repo**: [cookies-spec-issue](https://github.com/jennifer-mann/cookies-specs-issue)

**Comment by MartijnVanRiele** [comment]
Mornin’!

I run you repo and all the tests pass (unfortunately???).

We engaged the problem differently now by doing the actual login instead of setting cookies. This also makes the tests more robust for the futher.

However I also tried asserting your test and the all fail cause the actual cookie is different from the one that’s set in beforeEach….

Van: Jennifer Mann <notifications@github.com>
Beantwoorden - Aan: cypress-io/cypress <reply@reply.github.com>
Datum: donderdag 13 april 2017 23:14
Aan: cypress-io/cypress <cypress@noreply.github.com>
CC: "Riele, van, Martijn" <mvriele@Wehkamp.nl>, Author <author@noreply.github.com>
Onderwerp: Re: [cypress-io/cypress] Unable to run tests when they use different cookies (#484)


I was unable to recreate the issue above. I created a repo below that closely resembles the test code you described above. So, I have a few follow-up suggestions:

  1.  What happens if you write the assertion below in the first test of each file (substituting the values for the appropriate token & user-data), then click "Run All". Do these assertions pass? fail?

it('gets the browser cookie', function(){

    cy.getCookie('token').should('have.property', 'name', 'token')

    cy.getCookie('token').should('have.property', 'value', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJqdGkiOiIwNWE2YmY4OS0yZjE3LTRmZWYtYjI1Mi05NDAxY2ZjZWFlZmMiLCJhdXQiOiJmdWxsIiwicHJlX2F1dGhfc2hvcHBlciI6ImVlYmI0M2VlLTgzOWEtNDViNS05Njg1LTBmNDAyYzBjNDUxMCIsImlhdCI6MTQ5MDYwNTQwMiwic3ViIjoiMjQ1NjIzNjIiLCJpc3MiOiJibGF6ZS1nYXRld2F5IiwiaXNzX3JlYXNvbiI6ImxvZ2luIn0.acudOmfCFP5CzKAHD5KQYFeLFuBSoXDtstJgK11arqw')

    cy.getCookie('user-data').should('have.property', 'name', 'user-data')

    cy.getCookie('user-data').should('have.property', 'value', 'eyJhdXRoZW50aWNhdGlvbl90eXBlIjoiZnVsbCIsInRva2VuX2lkIjoiMDVhNmJmODktMmYxNy00ZmVmLWIyNTItOTQwMWNmY2VhZWZjIn0=')

})

  1.  Fork the repo below and get their tests to fail. There is something different that I haven't determined. This would be very helpful to us solving the issue.

Test Repo: cookies-spec-issue<https://github.com/jennifer-mann/cookies-specs-issue>
